src/sdl2/render.rs

```diff
@@ -492,30 +496,81 @@ impl TextureCreator {
     }
 }
 
+pub enum TargetKind {
+    Canvas,
+    Texture,
+}
+
+pub trait Target {
+    fn kind(&self) -> TargetKind;
+}
+
+pub struct CanvasTarget;
+
+impl Target for CanvasTarget {
+    fn kind(&self) -> TargetKind {
+        TargetKind::Canvas
+    }
+}
+
+pub struct TextureTarget<'r, 't> {
+    _renderer_marker: PhantomData<&'r ()>,
+    _texture_marker: PhantomData<&'t ()>,
+}
+
+impl<'r, 't> Target for TextureTarget<'r, 't> {
+    fn kind(&self) -> TargetKind {
+        TargetKind::Texture
+    }
+}
+
+pub type CanvasRenderer = Renderer<CanvasTarget>;
+
 /// Render target methods
-impl Renderer {
+impl CanvasRenderer {
     /// Determine whether a window supports the use of render targets.
     pub fn render_target_supported(&self) -> bool {
         unsafe { ll::SDL_RenderTargetSupported(self.context.raw) == 1 }
     }
 
-    /// Gets the render target handle.
-    ///
-    /// Returns `None` if the window does not support the use of render targets.
-    pub fn render_target(&self) -> Option<RenderTarget> {
+    /// Sets the render target to the provided texture
+    /// Returns a handle for rendering methods to the target
+    /// Returns `SdlError` if the window does not support the use of render targets,
+    /// or if SDL2 returned with an error code
+    /// The texture must be created with the texture access: `sdl2::render::TextureAccess::Target`.
+    pub fn target<'r, 't, 'a>(&'r mut self,
```

Comment:
`target` make it sound like you *return* a target instead of setting one. If you have a look at the current Renderer, to get the window you would call `window()` or `window_mut`, so `target` definitely feels like a getter and not a setter to me. How about `set_target` instead ?